src/fsharp/CompileOps.fs

```diff
@@ -823,10 +823,16 @@ let OutputPhasedErrorR (os: StringBuilder) (err: PhasedDiagnostic) (suggestNames
       | UndefinedName(_, k, id, suggestionsF) ->
           os.Append(k (DecompileOpName id.idText)) |> ignore
           if suggestNames then
-              let filtered = ErrorResolutionHints.FilterPredictions suggestionsF id.idText 
-              if List.isEmpty filtered |> not then
-                  os.Append(ErrorResolutionHints.FormatPredictions DecompileOpName filtered) |> ignore
-          
+              let buffer = ErrorResolutionHints.SuggestionBuffer(id.idText)
```

Comment:
Minor thing: Lines 826-835 and 1373-1382 are exactly the same. We should probably make a function to hold it in one one place. Perhaps:
```fsharp
          let suggestNames (os: StringBuilder) (text: string) =
              let buffer = ErrorResolutionHints.SuggestionBuffer(text)
              if not buffer.Disabled then
                  suggestionF buffer.Add
                  if not buffer.IsEmpty then
                      os.Append " " |> ignore
                      os.Append(FSComp.SR.undefinedNameSuggestionsIntro()) |> ignore
                      for value in buffer do
                          os.AppendLine() |> ignore
                          os.Append "   " |> ignore
                          os.Append(DecompileOpName value) |> ignore
```
above ```let rec OutputExceptionR (os: StringBuilder) error =```
and also change the `suggestNames: bool` paremeter on `OutputPhasedErrorR` to `canSuggestNames: bool`.

Reply:
ok